### ISSUE TITLE
Using ALVR with a USB-C Cable - Oculus Quest

### ISSUE THREAD

**Issue by bestgrafix** [polygraphene/ALVR#401]
Hey short question, would it be possible to use instead a wifi 5ghz connection the usb-c cable for steaming the content ?  Is that possible ? or is there a technical limitation, the latency should be lower and  the bitrate could be higher.

greetings

Ben

**Comment by babybasher** [comment]
I think the encoding/decoding time of the video is the limiting factor here.  The 'transport latency' is the lowest part of it all.

**Comment by bestgrafix** [comment]
This is the latency with 5g wireless, 20mbit, 150% Resolution, with cable it should much lower, or not ?!  Dont know, just thinking if you have with the quest a solution for both ...

![alvr_h264_150%_20mbit_5ghz](https://user-images.githubusercontent.com/35193059/59157542-bd830880-8aac-11e9-8572-9cc0890f2820.JPG)

**Comment by babybasher** [comment]
52ms is damn good.  But 20mb bitrate is really low, can't look great.  Maybe try upping it to 50mb, leave it at 150%.  

As you can see in your screen shot, the majority of the latency is encoding/decoding. And the higher you up the bitrate the longer that takes.  

It's just the way this method works.  I mean look at the VIVE wireless solution. They needa use a god damn 60GHZ wifi network to transmit that raw data.

**Comment by bestgrafix** [comment]
Yes thats what i mean , 20mbit is to low i find 30mbits ok but could be higher ... so why not not  use insead a 5m  usb-c cable directly ? That was my questions if this is possible ?

**Comment by babybasher** [comment]
Because the bottleneck isn't in the transport latency.  It's the time it takes your computer to encode/decode the video stream.  

For example.  The wifi chip in the Quest maxes at about 867megabits/sec.  

The max bitrate ALVR lets you do is 250Mbps.  Even if you had the worst AC wifi router at about 433Mbps that can still handle that (provided your PC is connected via ethernet cable to the router).

So the problem here isn't network bandwidth what so ever.  It's the processing power required to encode/decode the video.

**Comment by bonsaiheldin** [comment]
Isn't USB Tethering disabled on the Quest anyway? Oculus would have to activate it for us to use it. The only way i know of to stream over USB could be ABD tools, but i have no idea how that works and how fast it would be.

What's the use of more than 100% resolution if i may ask?

**Comment by jackfruh** [comment]
> 100% resolution is helpful with anti-aliasing decisions.
Basically, if you render at 100%, you're hitting peak efficiency, if your GPU has extra power to spare, you can render a higher resolution, then the system scales that down to fit the resolution of your screen.
In theory, this sounds like the end result would be no different.
In practice, there is quite a bit of extra information in the higher resolution and when the scaling algorithm lowers the resolution back down to 100% it (in 2019) makes different decisions than the original in game render and things like text often look much sharper.

In summary: Below 100% - good for slow systems, slow video cards - looks terrible
100% - the standard - good for system that can handle it - looks good.
> 100% - good for systems with ridiculous power - multi core with expensive graphics cards - gets the absolute most out of the resolution found in popular VR headsets of this day and age.

**Comment by safijari** [comment]
It all comes down to if the usb-c pipe is big enough to carry the full image without compression, and if the compute burden of doing so both on the server and the client would be lower than compression/decompression. If both those things are true then usb-c streaming would be viable.

@bestgrafix the screenshot you shared shows a 1ms total time for the actual transport (i.e. time taken for the frame to go from your computer to your headset). The absolute worst this time ever gets for me is about 10ms, that's the worst case scenario. The compression/encoding and decompression/decoding is what kills us here. As you increase the bitrate, those times also go up more so than the transport time. As an example: if I up the bitrate to 100mpbs, my transport time becomes something like 7ms, but time to encode and decode combined becomes 120ms. At 30mbps the encode/decode time combined ends up at 55ms with about 4ms for the transport latency. That's good enough for me for now, but we really need more efficient compression/decompression on both the host GPU as well as the quest.

**Comment by bestgrafix** [comment]
thanks for the explaination. For me it looks nvenc is used for encoding which is should be realy good. 

john carmack said on twitter:
We did not add any dedicated hardware to act as a PC display (it was debated a lot), but we have a research project going to see what we can do with maxing out WiFi streaming.  No promises...
